native-image-tests/src/main/scala/com/lightbend/EsbTester.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2009-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.pattern.StatusReply
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.serialization.jackson.JsonSerializable
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+
+import scala.concurrent.duration.DurationInt
+
+object EventSourcedCounter {
+  sealed trait Command extends JsonSerializable
+
+  final case class Increase(amount: Int, replyTo: ActorRef[StatusReply[Increased]]) extends Command
+  @JsonCreator
+  final case class GetValue(replyTo: ActorRef[StatusReply[GetValueResponse]]) extends Command
+  final case class GetValueResponse(value: Int)
+
+  sealed trait Event extends JsonSerializable
+
+  // FIXME why doesn't @JsonCreator work as usual? is it something missing from the akka jackson feature?
```

Comment:
`-parameters` java compiler flag is needed for Java, but I thought that wasn't needed for Scala case classes. Compare that with the other Scala samples we have.

Reply:
Investigations on the Akka side shows that the parameter names are in the reflection metadata, but it still just doesn't work for single param constructors without the `@JsonProperty` :/ 

Fix/investigation PR: https://github.com/akka/akka/pull/32326